src/Microsoft.AspNet.Razor/Parser/TagHelpers/TagHelperBlockRewriter.cs

```diff
@@ -28,7 +28,11 @@ public static class TagHelperBlockRewriter
             var attributes = GetTagAttributes(tagName, validStructure, tag, descriptors, errorSink);
             var selfClosing = IsSelfClosing(tag);
 
-            return new TagHelperBlockBuilder(tagName, selfClosing, start, attributes, descriptors);
+            // All descriptors for an element should share a common prefix, grab the first.
+            var prefix = descriptors.First().Prefix;
```

Comment:
neither `prefix` nor `prefix.Length` change for the lifetime of a `TagHelperParseTreeRewriter` instance.  calculate length once in that class's constructor and pass it into this method.